File: cpp-2021-prelim/p14.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    int x, y;
    cin >> x >> y;
    if (y % 2 || x * 4 < y || x * 2 > y) {
        cout << "No feasible solution";
        return 0;
    }
    cout << x * 2 - y / 2 << endl;
    cout << y / 2 - x;
}

File: rust-2021-prelim/p14.rs
use std::io::stdin;
use std::str::FromStr;

fn read<F: FromStr>() -> F {
    let mut input = String::new();
    stdin().read_line(&mut input).unwrap();
    input.trim().parse().unwrap()
}

fn main() {
    let x: i32 = read();
    let y: i32 = read();
    if (y % 2 != 0 || x * 2 > y || x * 4 < y) {
        println!("No feasible solution");
        return;
    }
    println!("{} {}", x * 2 - y / 2, y / 2 - x)
}
